src/components/Button.tsx

```diff
@@ -0,0 +1,98 @@
+import styled from "styled-components";
+
+interface ButtonProps {
+  size?: "small" | "medium" | "large";
+  variant?: "solid" | "outline" | "ghost" | "link";
+  backgroundColor?: string;
+  label: string;
+  onClick?: () => void;
+}
```

Comment:
```suggestion
interface ButtonProps extends Pick<CSSProperties, 'backgroundColor'> {
  size?: "small" | "medium" | "large";
  variant?: "solid" | "outline" | "ghost" | "link";
  label: string;
  onClick?: () => void;
}
```

P3:
React에서 제공하는 타입을 활용하면, style요소를 이렇게 타입 정의해 볼수있어요.
덧붙여서 DOM node property도 `HTMLProps<HTMLButtonElement>` 처럼 타입으로 정의 가능하답니다.